### PR TITLE
fix user locale handling 

### PR DESCRIPTION
**Pull request by rds13**
add more french translations
fix langage preference setup

### CODE REVIEW

File: locales/fr.js [hunk 1]
```diff
@@ -237,9 +237,10 @@
 	"logging_in": "Connexion",
 	"password_confirmation": "Confirmation du mot de passe",
 	"confirm_again": "Veuillez consulter votre boîte pour confirmer votre email.",
-	"confirmed": "Adresse email confirmée avec succès. merci!",
+	"confirmed": "Adresse email confirmée avec succès. Merci !",
+	"signing_up": "Signing Up",
 	"password_check_inbox": "password_check_inbox",
-	"what_is_your_name": "Bonjour! Choisir un nom d'utilisateur s'il vous plaît.",
+	"what_is_your_name": "Bonjour ! Choisir un nom d'utilisateur s'il vous plaît.",
```
Comment (mntmn) [inline locales/fr.js:243]: Can you explain why you put spaces before exclamation marks?

File: locales/fr.js [hunk 2]
```diff
@@ -271,18 +272,18 @@
...
+	"history_recently_updated": "Nouvelles",
+	"history_recently_empty": "Rien ne se passe",
 	"parent_folder": "Dossier origine",
 	"created_by": "Créé par",
 	"last_updated": "Mis à jour",
-	"history_recently_updated": "Nouvelles",
-	"history_recently_empty": "Rien ne se passe",
-	"signing_up": "Signing Up",
-	"feedback_sent": "Merci pour votre commentaire!",
+	"feedback_sent": "Merci pour votre commentaire !",
```
Comment (mntmn) [inline locales/fr.js:281]: Here too.

**Review by mntmn — CHANGES_REQUESTED** [review]
Thank you very much for this! I would like to know the rationale behind adding spaces in front of exclamation marks. Is that a common practice in french?

File: locales/fr.js (hunk 1 above)
Comment (rds13) [reply locales/fr.js:243]: It's the French convention for punctuation. Every punctuation with two items as ':', or '?' should preceded by a blank and followed by another blank.

**Review by mntmn — APPROVED** [review]

**Pull request merged**